File: leetcode/PathSum.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {

    public static boolean recCheckSum(TreeNode curr,  int accumulator, int targetSum) {

        if (curr == null) {
            return false;
        }

        accumulator += curr.val;

        if (accumulator == targetSum && curr.left == null && curr.right == null) {
            return true;
        }

        return recCheckSum(curr.left, accumulator, targetSum)  || recCheckSum(curr.right, accumulator, targetSum);

    }
    public boolean hasPathSum(TreeNode root, int targetSum) {

        if (root == null)  {
            return false;
        }

        if (root.left == null && root.right == null && targetSum == root.val) {
            return true;
        }

        return recCheckSum(root.left, root.val, targetSum)  || recCheckSum(root.right, root.val, targetSum);


    }
}

File: leetcode/PathSum.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @param {number} targetSum
 * @return {boolean}
 */

var checkSum = function(curr, accumulator, targetSum) {

    if (curr === null) {
        return false;
    }

    accumulator += curr.val;

    if (accumulator == targetSum && curr.left === null && curr.right === null) {
        return true;
    }

    return checkSum(curr.left, accumulator, targetSum) || checkSum(curr.right, accumulator, targetSum);


}

var hasPathSum = function(root, targetSum) {

    if (root === null) {
        return false;
    }

    if (root.val == targetSum && root.left === null && root.right === null){
        return true;
    }

    return checkSum(root.left, root.val, targetSum) || checkSum(root.right, root.val, targetSum);

};
